### ISSUE TITLE
1.0 release?

### ISSUE THREAD

**Issue by novemberborn** [evanw/node-source-map-support#223]
Shipping a 1.0 release will make it easier for different projects to end up with the same instance of `source-map-support` (by relying on dependency deduplication). This would lessen the impact of https://github.com/evanw/node-source-map-support/issues/200.

**Comment by LinusU** [comment]
While it would indeed be nice with an 1.0 version, wouldn't it just make that specific problem even worse? Instead of libraries being divided on `0.4.x` and `0.5.x`, there would then be 3 different ranges (`0.4.x`, `0.5.x`, `1.x.x`)?

Bumping `@babel/register` to `^0.5.0` would fix the problem with Babel and AVA?

**Comment by novemberborn** [comment]
> Bumping @babel/register to ^0.5.0 would fix the problem with Babel and AVA?

Yes, until there's a `0.6.0`. At least with a 1.0 release you can ship new features without requiring updates in dependents.

**Comment by LinusU** [comment]
Ah, I'm currently shipping both features and patches in the patch bump, and breaking changes in the minor bump, since that is how Npm seems to follow the 0.x semver range. So a `0.6.0` would be the equivalent to `2.0.0` if we switch to 1.0 now.

Still, I think 1.x is a good idea.

It could be nice to wait for `source-map` to release an 1.0 though, and then we could upgrade to that (which will be a breaking change anyhow) and release our 1.0.

**Comment by novemberborn** [comment]
> Still, I think 1.x is a good idea.

Yay!

> It could be nice to wait for source-map to release an 1.0 though, and then we could upgrade to that (which will be a breaking change anyhow) and release our 1.0.

The problem is that `source-map` has dropped support for Node.js 6, which isn't palatable to AVA or (I strongly suspect) Babel until April next year, when it loses LTS status.

Worst case we need to republish `source-map` `0.6` so both versions can be depended on, and then select the correct version based on the runtime.

**Comment by LinusU** [comment]
Hmm, just realised that we cannot upgrade source-map at all currently, since they only have an async api now 🤔

**Comment by novemberborn** [comment]
Oh dear.

It might be good to unite all source-map related packages into one organization, which could make it easier to evolve them with each others interests in mind. At this point I think `source-map` needs to be forked to have a synchronous implementation…

**Comment by joebowbeer** [comment]
This needs a 1.0.

None of the versions lower than one can be assumed to be compatible, so no hoisting is possible.

_To help developers who rely on your code, we recommend starting your package version at 1.0.0_
-[About Semantic Versioning](https://docs.npmjs.com/about-semantic-versioning)
